chasm/registry.go

```diff
@@ -33,30 +33,30 @@ import (
 
 var (
 	// This is golang type identifier regex.
-	nameValidator = regexp.MustCompile(`^[A-Za-z_][A-Za-z0-9_]*$`)
+	typeNameValidator = regexp.MustCompile(`^[A-Za-z_][A-Za-z0-9_]*$`)
 )
 
 type (
 	Registry struct {
-		componentByName map[string]*RegistrableComponent       // fully qualified name -> component
-		componentByType map[reflect.Type]*RegistrableComponent // component type -> component
+		componentByType   map[string]*RegistrableComponent       // fully qualified type name -> component
+		componentByGoType map[reflect.Type]*RegistrableComponent // component go type -> component
 
-		taskByName map[string]*RegistrableTask       // fully qualified name -> task
-		taskByType map[reflect.Type]*RegistrableTask // task type -> task
+		taskByType   map[string]*RegistrableTask       // fully qualified type name -> task
+		taskByGoType map[reflect.Type]*RegistrableTask // task go type -> task
 	}
 )
 
 func NewRegistry() *Registry {
 	return &Registry{
-		componentByName: make(map[string]*RegistrableComponent),
-		componentByType: make(map[reflect.Type]*RegistrableComponent),
-		taskByName:      make(map[string]*RegistrableTask),
-		taskByType:      make(map[reflect.Type]*RegistrableTask),
+		componentByType:   make(map[string]*RegistrableComponent),
+		componentByGoType: make(map[reflect.Type]*RegistrableComponent),
+		taskByType:        make(map[string]*RegistrableTask),
+		taskByGoType:      make(map[reflect.Type]*RegistrableTask),
 	}
 }
 
 func (r *Registry) Register(lib Library) error {
-	if err := r.validateName(lib.Name()); err != nil {
+	if err := r.validateTypeName(lib.Name()); err != nil {
```

Comment:
:( It's still library.Name().library type will be weird I think.

Maybe I should use `typeName` everywhere. That's also what  golang does. There's a Name() method  in reflect.Type.

Reply:
I think it is fine. Go package has name, types inside it also have names. FQN on usage size looks like `<package_name>.<type_name>`. I would revert `validateTypeName` back to `validateName` because it validates all different names, not only types. Same for `typeNameValidator` var.

Reply:
Done. Reverted back to `nameValidator` and `validateName`.